### PR TITLE
View Engine - Change Detection Features

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/@angular/core/src/view/pure_expression.ts [hunk 1]
```diff
@@ -121,7 +132,7 @@ export function checkAndUpdatePureExpressionInline(
           case 4:
             value[3] = v3;
           case 3:
-            value[2] = v2;
```
Comment (tbosch) [inline modules/@angular/core/src/view/pure_expression.ts]: undo!

File: modules/@angular/core/src/view/pure_expression.ts [hunk 2]
```diff
@@ -208,16 +219,23 @@ export function checkAndUpdatePureExpressionDynamic(view: ViewData, def: NodeDef
...
+          value[bindings[i].name] = unwrapValue(values[i]);
         }
         break;
       case PureExpressionType.Pipe:
-        value = data.pipe.transform(values[0], ...values.slice(1));
+        const params = new Array(values.length);
+        for (let i = 0; i < values.length; i++) {
+          params[i] = unwrapValue(values[i]);
+        }
+        value = data.pipe.transform(params[0], ...params.slice(1));
```
Comment (tbosch) [inline modules/@angular/core/src/view/pure_expression.ts]: just use `...params`

File: modules/@angular/core/test/view/component_view_spec.ts [hunk 3]
```diff
@@ -69,65 +70,212 @@ function defineTests(config: {directDom: boolean, viewFlags: number}) {
...
+                  update)),
+        ]));
+
+        const compView = asProviderData(view, 1).componentView;
+
+        update.and.callFake((view: ViewData) => {
+          setCurrentNode(view, 0);
+          throw new Error('Test');
+        });
+        try {
```
Comment (tbosch) [inline modules/@angular/core/test/view/component_view_spec.ts]: change: `expect(..).toThrow()`

File: modules/@angular/core/test/view/component_view_spec.ts [hunk 4]
```diff
@@ -69,65 +70,212 @@ function defineTests(config: {directDom: boolean, viewFlags: number}) {
...
+          checkAndUpdateView(view);
+        } catch (e) {
+        }
+        expect(update).toHaveBeenCalled();
+
+        update.calls.reset();
+        update.and.returnValue(null);
```
Comment (tbosch) [inline modules/@angular/core/test/view/component_view_spec.ts]: remove

File: modules/@angular/core/src/view/pure_expression.ts (hunk 1 above)
Comment (tbosch) [inline modules/@angular/core/src/view/pure_expression.ts]: Done

File: modules/@angular/core/src/view/pure_expression.ts (hunk 2 above)
Comment (tbosch) [inline modules/@angular/core/src/view/pure_expression.ts]: Done

File: modules/@angular/core/test/view/component_view_spec.ts (hunk 3 above)
Comment (tbosch) [inline modules/@angular/core/test/view/component_view_spec.ts]: Done

File: modules/@angular/core/test/view/component_view_spec.ts (hunk 4 above)
Comment (tbosch) [inline modules/@angular/core/test/view/component_view_spec.ts]: Done

**Review by vicb — APPROVED** [review]

**Comment by tbosch** [comment]
@mhevery This marker code-review/pullapprove wrong (@vicb approved it) , @IgorMinar will look into it...

**Review by mhevery — APPROVED** [review]

**Pull request closed without merging**